spec/widgets/auto-style.spec.js

```diff
@@ -0,0 +1,30 @@
+var specHelper = require('../spec-helper');
```

Comment:
Now that we have some nice subclasses of the different types of auto stylers, I think it makes sense to actually unit test each type and make sure that `.getStyle` generates the stuff right. That's where things can go wrong, so that's the method I would unit test. Here's a suggestion of how I'd structure a test for the category auto styler:

```
var CategoryAutoStyler = require('../../auto-style/category.js')

describe('src/widgets/auto-style/category', function () {
  beforeEach(function () {
    this.dataview = new Backbone.Model({
      allCategoryNames: 'a', 'b', 'c',
      column: 'something' 
    })

    this.layer = this.dataview.layer = jasmine.createSpyObj('layer', ['getGeometryType']);
    this.categoryAutoStyler = new CategoryAutoStyler({
      dataview: this.dataview
    });
  });

  describe('.getStyle', function () {
    it('should generate the right styles when layer has polygons', function () {
      this.layer.getGeometryType.and.returnValue('polygon');

      expect(this.categoryAutoStyler.getStyle()).toEqual('EXPECTED_STYLE');
    })

    it('should generate the right styles when layer has points', function () {
      this.layer.getGeometryType.and.returnValue('point');

      expect(this.categoryAutoStyler.getStyle()).toEqual('EXPECTED_STYLE');
    })

    it('should generate the right styles when layer has lines', function () {
      this.layer.getGeometryType.and.returnValue('line');

      expect(this.categoryAutoStyler.getStyle()).toEqual('EXPECTED_STYLE');
    })

    it('should generate the right styles when layer\'s geometry cannot be infered', function () {
      this.layer.getGeometryType.and.returnValue(undefined);

      expect(this.categoryAutoStyler.getStyle()).toEqual('EXPECTED_STYLE');
    })
  });
});
```